crates/api_models/src/payment_methods.rs

```diff
@@ -856,6 +874,36 @@ pub struct PaymentMethodResponse {
     pub client_secret: Option<String>,
 }
 
+#[cfg(all(feature = "v2", feature = "payment_methods_v2"))]
+#[derive(Debug, serde::Deserialize, serde::Serialize, ToSchema, Clone)]
+pub struct ConnectorTokenDetails {
+    /// The unique identifier of the connector account through which the token was generated
+    #[schema(value_type = String, example = "mca_")]
+    pub connector_id: id_type::MerchantConnectorAccountId,
+
+    #[schema(value_type = TokenizationType)]
+    pub token_type: common_enums::TokenizationType,
+
+    /// The status of connector token
+    #[schema(value_type = ConnectorMandateStatus)]
+    pub status: common_enums::ConnectorMandateStatus,
+
+    /// The mandate reference id of the connector token
+    pub connector_token_request_reference_id: Option<String>,
+
+    pub original_payment_authorized_amount: Option<MinorUnit>,
+
+    /// The currency of the original payment authorized amount
+    #[schema(value_type = Currency)]
+    pub original_payment_authorized_currency: Option<common_enums::Currency>,
+
+    /// Metadata associated with the connector token
+    pub metadata: Option<pii::SecretSerdeValue>,
+
+    /// The value of the connector token
```

Comment:
Can you add more context toward `token` and `connector_token_request_reference_id`. And in the comment we are using mandate which we can avoid, this adds more terms and which terms represent what is slightly not clear.